### ISSUE TITLE
NSFS | S3 | Versioning: InternalError on DeleteObject in version-suspended mode

### ISSUE THREAD

**Issue by hseipp** [noobaa/noobaa-core#8382]
### Environment info
- NooBaa Version: **5.17.0-20240904**
- Platform: **CentOS8, deployed using [https://github.com/IBM/StorageScaleVagrant](https://github.com/IBM/StorageScaleVagrant), Scale 5.2.2.0 240913.105913**

### Actual behavior

DeleteObject returns InternalError:

```
    def test_s3cever104():
        client = get_client()
        bucket_name = get_new_bucket(client=client)
    
        # The following operation uses another client instance -> following operations will fail
    #    check_configure_versioning_retry(bucket_name, "Enabled", "Enabled")
        _ces_safe_check_configure_versioning_retry(client, bucket_name, "Enabled", "Enabled")
    
        key1_name = "test"
        version_ids = []
        contents = []
        num_versions = 3
        (version_ids, contents) = create_multiple_versions(client, bucket_name, key1_name, num_versions)
    
        _ces_safe_check_configure_versioning_retry(client, bucket_name, "Suspended", "Suspended")
    
        key2_name = 'obj2'
        data = 'object2'
        response = client.put_object(Bucket=bucket_name, Key=key2_name, Body=data)
        version_id_2 = None
        try:
            version_id_2 = response['VersionId']
        except KeyError:
            version_id_2 = None
        assert version_id_2 == None
    
>       client.delete_object(Bucket=bucket_name, Key=key2_name)

s3tests_boto3/functional/test_s3.py:2159: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.tox/py/lib/python3.8/site-packages/botocore/client.py:569: in _api_call
    return self._make_api_call(operation_name, kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <botocore.client.S3 object at 0x7f57d225adf0>, operation_name = 'DeleteObject', api_params = {'Bucket': 's3tests-qa4urc0y2bi1z8kt5a1nv-1', 'Key': 'obj2'}
...
E           botocore.exceptions.ClientError: An error occurred (InternalError) when calling the DeleteObject operation (reached max retries: 4): We encountered an internal error. Please try again.
```

### Expected behavior

DeleteObject should pass.

### Steps to reproduce

Execute the following test as addition to Ceph s3-tests:

```
def test_s3cever104():
    client = get_client()
    bucket_name = get_new_bucket(client=client)

    # The following operation uses another client instance -> following operations will fail
#    check_configure_versioning_retry(bucket_name, "Enabled", "Enabled")
    _ces_safe_check_configure_versioning_retry(client, bucket_name, "Enabled", "Enabled")

    key1_name = "test"
    version_ids = []
    contents = []
    num_versions = 3
    (version_ids, contents) = create_multiple_versions(client, bucket_name, key1_name, num_versions)

    _ces_safe_check_configure_versioning_retry(client, bucket_name, "Suspended", "Suspended")

    key2_name = 'obj2'
    data = 'object2'
    response = client.put_object(Bucket=bucket_name, Key=key2_name, Body=data)
    version_id_2 = None
    try:
        version_id_2 = response['VersionId']
    except KeyError:
        version_id_2 = None
    assert version_id_2 == None

    client.delete_object(Bucket=bucket_name, Key=key2_name)
    
    response = client.list_object_versions(Bucket=bucket_name)
    versions = response['Versions']
    assert len(versions) == 3
    delete_markers = response['DeleteMarkers']
    assert len(delete_markers) == 1

    client.delete_object(Bucket=bucket_name, Key=key1_name)

    response = client.list_object_versions(Bucket=bucket_name)
    versions = response['Versions']
    assert len(versions) == 3
    delete_markers = response['DeleteMarkers']
    assert len(delete_markers) == 2

    client.delete_object(Bucket=bucket_name, Key=key1_name, VersionId=version_ids[0])

    response = client.list_object_versions(Bucket=bucket_name)
    versions = response['Versions']
    assert len(versions) == 2
    delete_markers = response['DeleteMarkers']
    assert len(delete_markers) == 3
```

### More information - Screenshots / Logs / Other output

Noobaa log with "all" log level attached
[noobaa_20240918_3.log.gz](https://github.com/user-attachments/files/17042444/noobaa_20240918_3.log.gz)

**Comment by nadavMiz** [comment]
seems to be the same issue fixed by #8345 :
`Sep 17 17:00:56 m1.example.com node[19062]: Sep-17 17:00:56.063 [nsfs/19139] [ERROR] core.util.native_fs_utils:: safe_unlink: dir_file is undefined, cannot use it to call safe_unlink_gpfs`

However testing with the fix, a different error seems to happen:
```
        response = client.list_object_versions(Bucket=bucket_name)
        versions = response['Versions']
        assert len(versions) == 2
        delete_markers = response['DeleteMarkers']
>       assert len(delete_markers) == 3
E       AssertionError: assert 2 == 3
E        +  where 2 = len([{'IsLatest': True, 'Key': 'obj2', 'LastModified': datetime.datetime(2024, 9, 19, 8, 26, 10, tzinfo=tzutc()), 'Owner':...24, 9, 19, 8, 26, 10, tzinfo=tzutc()), 'Owner': {'DisplayName': 's3user-5002', 'ID': '66ebced60984526d0df35ee1'}, ...}])

```

**Comment by nadavMiz** [comment]
@hseipp the new error seems to be a problem with the test deleting specific version should not create a new delete marker.
changing the expected delete marker to 2, the test seems to pass

**Comment by hseipp** [comment]
@nadavMiz Thank you for spotting the error in the test code, the expected delete marker count should indeed be 2. Fix is pushed to the test code repo.

**Comment by nadavMiz** [comment]
Closing as Fixed by #8345
